vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceOverviewManager.java

```diff
@@ -0,0 +1,298 @@
+
+    /**
+     * To retrieve all space to space and space to module links corresponding to
+     * each space
+     * 
+     * @param contextPath           The contextpath of the application is being
+     *                              passed
+     * @param mapper                Passing this object to create object node for
+     *                              creating json of space and module node in the
+     *                              spaceoverview graph
+     * @param nodeArray             To create the json for each node of
+     *                              spaceoverview graph
+     * @param spaceLinkMap          For holding list of all links corresponding to
+     *                              each space
+     * @param spaceToModuleLinksMap For getting list of links from one space to
+     *                              other modules
+     * @param spaceToSpaceLinksMap  For getting list of links from one space to
+     *                              other spaces
+     */
+    @Override
+    public void getSpaceLinkMap(String contextPath, ObjectMapper mapper, ArrayNode nodeArray,
+            Map<String, List<String>> spaceLinkMap, Map<String, List<ModuleLinkDisplay>> spaceToModuleLinksMap,
+            Map<String, List<SpaceLinkDisplay>> spaceToSpaceLinksMap) {
+
+        Iterable<Space> allSpacesList = spaceRepo.findAll();
+        if (allSpacesList != null) {
+            createSpaceNodeInOverviewGraph(contextPath, mapper, nodeArray);
+            for (Space space : allSpacesList) {
+                List<ModuleLinkDisplay> spaceToModulelinksList = spaceToModuleLinksMap.get(space.getId());
+                List<SpaceLinkDisplay> spaceLinks = spaceToSpaceLinksMap.get(space.getId());
+                List<SpaceLinkDisplay> spaceToSpaceLinksList = new ArrayList<>();
+                if (spaceLinks != null) {
+                    spaceToSpaceLinksList.addAll(spaceLinks);
+                }
+                List<String> spaceAllLink = new ArrayList<>();
+
+                if (spaceToModulelinksList != null) {
+                    spaceToModulelinksList.forEach(link -> {
+                        if (link.getLink() != null && link.getLink().getModule() != null
+                                && link.getLink().getModule().getId() != null) {
+                            spaceAllLink.add(link.getLink().getModule().getId());
+                        }
+                    });
+                }
+                if (spaceToSpaceLinksList != null) {
+                    spaceToSpaceLinksList.forEach(link -> {
+                        if (link.getLink() != null && link.getLink().getTarget() != null
+                                && link.getLink().getTarget().getId() != null) {
+                            spaceAllLink.add(link.getLink().getTarget().getId());
+                        }
+                    });
+                }
+                if (space.getId() != null) {
+                    spaceLinkMap.put(space.getId(), spaceAllLink);
```

Comment:
unless there is a good reason why `spaceLinkMap` is passed into the method,  it should be created inside this method and then returned.